Cargo.toml

```diff
@@ -17,23 +17,32 @@ virtualbox = ["fdp"]
 # Hyper-v driver
 hyper-v = ["winapi", "widestring", "ntapi", "vid-sys"]
 
-
 [dependencies]
 log = "0.4.8"
 libc = { version = "0.2.58", optional = true }
 xenctrl = { git = "https://github.com/Wenzel/xenctrl", optional = true }
 xenstore = { git = "https://github.com/Wenzel/xenstore", optional = true }
 xenforeignmemory = { git = "https://github.com/Wenzel/xenforeignmemory", optional = true }
-kvmi = { git = "https://github.com/Wenzel/kvmi", optional = true }
 fdp = { git = "https://github.com/Wenzel/fdp", optional = true }
+kvmi = { git = "https://github.com/arnabcs17b006/kvmi", optional = true }
 winapi = { version = "0.3.8", features = ["tlhelp32", "winnt", "handleapi", "securitybaseapi"], optional = true }
 widestring = { version = "0.4.0", optional = true }
 ntapi = { version = "0.3.3", optional = true }
 vid-sys = { version = "0.3.0", features = ["deprecated-apis"], optional = true }
 cty = "0.2.1"
+enum-primitive-derive = "0.1.2"
+mockall = "0.7.1"
+kvmi-sys = { git = "https://github.com/Wenzel/kvmi-sys.git" }
 
 [dev-dependencies]
 env_logger = "0.7.1"
 ctrlc = "3.1.3"
 clap = "2.33.0"
 colored = "1.9.3"
+
```

Comment:
A side note: Throughout your code I've seen several occasions where you insert or delete lines without a specific reason. It would be nice if you avoided such changes. In this particular case it helps to active the setting "Trim Final Newlines" in vscode for example.